executionlayer/execution-layer_test.go

```diff
@@ -1433,3 +1481,92 @@ func TestELHandleSubscriptionError(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestValidateEIP1271(t *testing.T) {
+	et := setup(t, &happyEC{t,
+		[]*mockNode{
+			&mockNode{
+				addr:      common.HexToAddress("0x0000000000000000000001234567899876543210"),
+				inSP:      true,
+				minipools: 1,
+			},
+		},
+		[]*mockNode{
+			&mockNode{
+				addr:      common.HexToAddress("0x0000000000222222222222222222222222222222"),
+				inSP:      false,
+				minipools: 0,
+			},
+		},
+	})
+
+	if err := et.ec.Init(); err != nil {
+		t.Fatal(err)
+	}
+
+	errs := make(chan error)
+	go func() {
+		if err := et.ec.Start(); err != nil {
+			errs <- err
+		}
+		close(errs)
+	}()
+
+	// Wait for connection
+	<-et.ec.connected
+
+	// Test cases
+	testCases := []struct {
+		name           string
+		dataHash       [32]byte
+		signature      []byte
+		address        common.Address
+		expectedResult bool
+		expectedError  bool
+	}{
+		{
+			name:           "Valid signature",
+			dataHash:       [32]byte{0x00},
+			signature:      common.FromHex(eip1271ValidSignature),
+			address:        common.HexToAddress(eip1271SmartContractValidSignerAddress),
+			expectedResult: true,
+			expectedError:  false,
+		},
+		{
+			name:           "Invalid signature",
+			dataHash:       [32]byte{0x00},
+			signature:      common.FromHex(eip1271InvalidSignature),
+			address:        common.HexToAddress(eip1271SmartContractValidSignerAddress),
+			expectedResult: false,
+			expectedError:  false,
+		},
+		// TODO: add test for when smart contract call reverts (like a contract that doesn't implement the isValidSignature method)
```

Comment:
@jshufro I could use some assistance here with getting some more non-happy-path tests written, like for if the given contract doesn't exist or doesn't implement the `isValidSignature` function (causing a revert)